tests/unit/tools/test_exception_run.py

```diff
@@ -0,0 +1,146 @@
+import os
+from pathlib import Path
+from typing import Any
+from unittest.mock import patch
+
+import pytest
+from litellm.types.utils import (
+    ChatCompletionMessageToolCall,
+    Choices,
+    Delta,
+    Function,
+    Message,
+    ModelResponse,
+    StreamingChoices,
+)
+
+from any_agent import (
+    AgentConfig,
+    AgentFramework,
+    AgentRunError,
+    AnyAgent,
+)
+from any_agent.tracing.otel_types import StatusCode
+
+
+def test_exception_trace(
+    agent_framework: AgentFramework,
+    patched_function: str,
+    tmp_path: Path,
+) -> None:
+    if agent_framework in (
+        # Fails due to:
+        # `('Recursion limit of 25 reached without hitting a stop condition. You can increase the ...oubleshooting/errors/GRAPH_RECURSION_LIMIT',)`
+        AgentFramework.LANGCHAIN,
+    ):
+        pytest.skip(f"Framework {agent_framework.value} not currently passing the test")
+
+    kwargs = {}
+
+    tmp_file = "tmp.txt"
+    # FIXME patch some call so an exception is triggered
+    # An exception within a tool will be handled by the framework
```

Comment:
Can this comment be removed and either the fixme addressed, or a Github issue created to track this, if it's an issue that should be solved